### PR TITLE
Test/integration tests

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?
https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/bookstore
Because the book index page needed more test coverage
Because the controller tests were redundant and fragile

### What Changed?
Added book index page integration tests
Added instance variables to tests for data reusability
Removed redundant/fragile controller tests

### CODE REVIEW

File: features/scenarios/book_index_page.feature [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    Then I see a list of books in the database
+      And the books are ordered by published date
+      And the list of books is paginated 25 books per page
+
+  Scenario: Book searching
+    Given there are 100 books in the database
+      And I am logged into the site
+    When I visit the root url
+      And I enter a book's title into the book search field
+    Then I am shown a list of books with that title
```
Comment (jahammo2) [inline features/scenarios/book_index_page.feature]: Good

**Comment by jahammo2** [comment]
Why did you think that your controller questions were fragile?

**Comment by TylerRockwell** [comment]
Anytime I changed things regarding sort order and pagination, it caused the controller index test to fail. This functionality is now being covered in the integration test, so rather than taking time to account for these scenarios, I dropped the controller tests in favor of the integration tests.

**Comment by jahammo2** [comment]
Fair enough. I can't say for sure if that was the right solution but adding more tests is almost always good. :shipit:

**Comment by TylerRockwell** [comment]
I'll ask about it later today when I talk with Annie and get her opinion on it. I'm not certain where the threshold of extra testing vs. maintenance time lies. Thanks for the second review in 2 hours, @jahammo2 !

**Pull request merged**